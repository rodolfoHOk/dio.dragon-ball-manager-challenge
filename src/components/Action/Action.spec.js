import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';
import Action from './Action';

import profileJson from '../../mocks/profile.json';
import profileSuccess from '../../mocks/profileSuccess.json';

it('Should show modal when try invoke', async () => {
  const { container, getByText } = render(
    <Action balls={profileJson.profile.balls} />
  );

  const button = getByText('Invocar').closest('button');
  fireEvent.click(button);

  expect(
    getByText('Você não tem todas as esferas para invocar o shenlong')
  ).toBeInTheDocument();

  const backButton = getByText('Voltar').closest('button');
  fireEvent.click(backButton);

  expect(getByText('Invocar shenlong')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('Should show shenlong when invoke', async () => {
  const { getByText, getByTestId } = render(
    <Action balls={profileSuccess.profile.balls} />
  );

  const button = getByText('Invocar').closest('button');
  fireEvent.click(button);

  expect(getByTestId('shenlong')).toBeInTheDocument();
});
