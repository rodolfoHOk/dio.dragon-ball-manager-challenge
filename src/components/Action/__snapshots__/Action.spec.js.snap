// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should show modal when try invoke 1`] = `
<div>
  <div
    class="sc-aXZVg gSWiUs"
    data-testid="card-shenlong"
  >
    <div
      class="css-1brk784"
    >
      Invocar shenlong
    </div>
    <button
      class="btn btn-warning"
      data-testid="invoke-button"
      type="button"
    >
      Invocar
    </button>
  </div>
</div>
`;
